extensions/transferdetails/service.go

```diff
@@ -0,0 +1,272 @@
+	if err != nil {
+		return nil, errors.NewTypedError(documents.ErrDocumentInvalid, err)
+	}
+
+	return model, nil
+}
+
+// DeriveFromUpdatePayload derives Funding from clientUpdatePayload
+func (s service) DeriveFromUpdatePayload(ctx context.Context, req UpdateTransferDetailRequest, identifier []byte) (documents.Model, error) {
+	model, err := s.GetCurrentVersion(ctx, identifier)
+	if err != nil {
+		transfersLog.Error(err)
+		return nil, documents.ErrDocumentNotFound
+	}
+
+	idx, err := extensions.FindAttributeSetIDX(model, req.TransferId, transfersLabel, transferIDLabel, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := deriveDIDs(req.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	// overwriting is not enough because it is not required that
+	// the funding payload contains all funding attributes
+	model, err = extensions.DeleteAttributesSet(model, TransferDetailData{}, idx, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	attributes, err := extensions.FillAttributeList(*req.Data, idx, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	err = model.AddAttributes(
+		documents.CollaboratorsAccess{
+			ReadWriteCollaborators: c,
+		},
+		true,
+		attributes...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	validator := CreateValidator()
+	err = validator.Validate(nil, model)
+	if err != nil {
+		return nil, errors.NewTypedError(documents.ErrDocumentInvalid, err)
+	}
+
+	return model, nil
+}
+
+// TODO: generic?
+func (s service) findTransfer(model documents.Model, transferID string) (*TransferDetailData, error) {
+	idx, err := extensions.FindAttributeSetIDX(model, transferID, transfersLabel, transferIDLabel, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+	return s.deriveTransferData(model, idx)
+}
+
+// TODO: generic?
+func (s service) deriveTransferData(model documents.Model, idx string) (*TransferDetailData, error) {
+	data := new(TransferDetailData)
+
+	types := reflect.TypeOf(*data)
+	for i := 0; i < types.NumField(); i++ {
+		// generate attr key
+		jsonKey := types.Field(i).Tag.Get("json")
+		label := extensions.LabelFromJSONTag(idx, jsonKey, transfersFieldKey)
+
+		attrKey, err := documents.AttrKeyFromLabel(label)
+		if err != nil {
+			return nil, err
+		}
+
+		if model.AttributeExists(attrKey) {
+			attr, err := model.GetAttribute(attrKey)
+			if err != nil {
+				return nil, err
+			}
+
+			// set field in data
+			n := types.Field(i).Name
+
+			v, err := attr.Value.String()
+			if err != nil {
+				return nil, err
+			}
+
+			reflect.ValueOf(data).Elem().FieldByName(n).SetString(v)
+
+		}
+
+	}
+	return data, nil
+}
+
+// DeriveTransferResponse returns create response from the added transfer detail
+func (s service) DeriveTransferResponse(ctx context.Context, model documents.Model, transferID string) (*TransferDetailResponse, error) {
+	idx, err := extensions.FindAttributeSetIDX(model, transferID, transfersLabel, transferIDLabel, transfersFieldKey)
```

Comment:
same here:

the method could be moved to the extension package together with an extra parameter for the struct.

this method would only call the generic method and afterwards return the `&TransferDetailResponse{...}`